core/tests/puppeteer-acceptance-tests/spec/exploration-editor-tests/save-draft-publish-and-discard-the-changes.spec.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for saving drafts, publishing, and discarding changes.
+ */
+
+import {showMessage} from '../../puppeteer-testing-utilities/show-message-utils';
+import testConstants from '../../puppeteer-testing-utilities/test-constants';
+import {UserFactory} from '../../puppeteer-testing-utilities/user-factory';
+import {ExplorationEditor} from '../../user-utilities/exploration-editor-utils';
+
+const DEFAULT_SPEC_TIMEOUT_MSECS = testConstants.DEFAULT_SPEC_TIMEOUT_MSECS;
+enum INTERACTION_TYPES {
+  END_EXPLORATION = 'End Exploration',
+}
+
+describe('Exploration Creator', function () {
+  let explorationEditor: ExplorationEditor;
+  let explorationVisitor: ExplorationEditor;
```

Comment:
Why is the visitor an "exploration editor"? Shouldn't they just be a regular user?

Reply:
Done